R/zchunk_L1233.Elec_water.R

```diff
@@ -39,89 +38,301 @@ module_water_L1233.Elec_water_DISABLED <- function(command, ...) {
     A23.CoolingSystemShares_RG3 <- get_data(all_data, "water/A23.CoolingSystemShares_RG3")
     elec_tech_water_map <- get_data(all_data, "water/elec_tech_water_map")
     Macknick_elec_water_m3MWh <- get_data(all_data, "water/Macknick_elec_water_m3MWh")
 
+    L101.en_bal_EJ_ctry_Si_Fi_Yh_full <-
+      get_data(all_data, "temp-data-inject/L101.en_bal_EJ_ctry_Si_Fi_Yh_full") %>%
+      gather(year, value, -iso, -GCAM_region_ID, -sector, -fuel) %>%
+      mutate(year = as.integer(substr(year, 2, 5)))
+
+    L1231.in_EJ_R_elec_F_tech_Yh <-
+      get_data(all_data, "temp-data-inject/L1231.in_EJ_R_elec_F_tech_Yh") %>%
+      gather(year, value, -GCAM_region_ID, -sector, -fuel, -technology) %>%
+      mutate(year = as.integer(substr(year, 2, 5)))
+
+    L1231.out_EJ_R_elec_F_tech_Yh <-
+      get_data(all_data, "temp-data-inject/L1231.out_EJ_R_elec_F_tech_Yh") %>%
+      gather(year, value, -GCAM_region_ID, -sector, -fuel, -technology) %>%
+      mutate(year = as.integer(substr(year, 2, 5)))
+
     # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses repeat_and_add_vector
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
 
-    # Produce outputs
-    # Temporary code below sends back empty data frames marked "don't test"
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    # STEP 1. DOWNSCALE WATER COOLING SYSTEM SHARES FROM 14 REGIONS to 201 COUNTRIES
+
+    # RENAME INTERMEDIATE FUELS AND AGGREGATE BY UPDATED FUEL TYPE
+    L101.en_bal_EJ_ctry_Si_Fi_Yh_full %>%
+      filter(sector == "out_electricity generation") %>%
+      left_join(select(enduse_fuel_aggregation, fuel, electricity), by = "fuel") %>%
```

Comment:
Actually this should technically be a restricted join because there are matching entries for all fuels. The problem is some of the entries in the file being joined are already labelled NA (i.e., before the join). The `left_join_error_no_match` function sees the NAs in the output and assumes incorrectly that these are failed matches... I'm wondering if `left_join_error_no_match` should be relaxed to allow for this type of case.